### PR TITLE
Portal 1438/file submission modal

### PR DESCRIPTION
**Pull request by ZacharyRizer**
Related to this ticket: https://cdc-dexops.atlassian.net/browse/PORTAL-1438

* added DetailsModal component
* incorporated new components and icons (ProgressTracker and Accordion)
* added api call for submissionDetails and tests
* the pill icons were added to the pill in the details modal but also in the StatusBoxes and PieChart

### CODE REVIEW

File: src/components/DetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    const fetchCall = async () => {
+      const res = await getSubmissionDetails(
+        auth.user?.access_token || "",
+        submission.upload_id
+      );
+
+      if (res.status != 200) return;
+
+      try {
+        // TODO: replace any with correct object structure
```
Comment (tbolt) [inline src/components/DetailsModal.tsx:51]: This can probably come out since it's mocked.

File: src/components/DetailsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  };
+
+  return (
+    <Modal
+      isOpen={isModalOpen}
+      modalTitle="Submission Details"
+      onClose={handleModalClose}>
+      <ModalBody>
+        <span className="padding-bottom-3">
+          {/* TODO: update cdc-react to allow className on <Pill> so we can avoid this extra span */}
```
Comment (tbolt) [inline src/components/DetailsModal.tsx:107]: I'll take care of this in my ticket but leave this TODO for now 👍

File: src/components/DetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        </span>
+        <div className="grid-row flex-row flex-align-center padding-bottom-3">
+          <Icons.PaperLines />
+          <h2 className="font-sans-md text-bold padding-left-1">
+            {submission.filename}
+          </h2>
+        </div>
+        {getContent()}
+        <Divider className="margin-y-2" height={4} stroke="#E0E0E0" />
+        <div className="grid-row margin-y-1">
```
Comment (tbolt) [inline src/components/DetailsModal.tsx:124]: Nice, I haven't used the grid- utility classes yet

File: src/screens/Submissions.tsx [hunk 4]
```diff
@@ -133,36 +139,45 @@ function Submissions() {
...
+                      label={getStatusDisplayValuesById(item.status).label}
+                      color={getStatusDisplayValuesById(item.status).pillColor}
                     />
                   </TableDataCell>
                   <TableDataCell size="md">
                     {new Date(item.timestamp).toLocaleString()}
                   </TableDataCell>
                   <TableDataCell size="sm" className="details-row">
-                    <Icons.Dots />
+                    {/* Todo: Instead of using a span here, we should update Icons to allow an onClick */}
```
Comment (tbolt) [inline src/screens/Submissions.tsx:156]: This comment can come out since we have a button here now. Good update

**Review by tbolt — APPROVED** [review]
Looks good, nice work

File: src/components/DetailsModal.tsx (hunk 1 above)
Comment (ZacharyRizer) [reply src/components/DetailsModal.tsx]: Agreed, removed.

File: src/components/DetailsModal.tsx (hunk 2 above)
Comment (ZacharyRizer) [reply src/components/DetailsModal.tsx:100]: Sounds good.

File: src/screens/Submissions.tsx (hunk 4 above)
Comment (ZacharyRizer) [reply src/screens/Submissions.tsx:156]: Removed

**Pull request merged**